pyquil/paulis.py

```diff
@@ -1133,18 +1105,19 @@ def suzuki_trotter(trotter_order: int, trotter_steps: int) -> List[Tuple[float,
 
 
 def is_zero(pauli_object: PauliDesignator) -> bool:
-    """
-    Tests to see if a PauliTerm or PauliSum is zero.
+    """Return True if a PauliTerm or PauliSum is zero, False otherwise.
 
     :param pauli_object: Either a PauliTerm or PauliSum
     :returns: True if PauliTerm is zero, False otherwise
     """
     if isinstance(pauli_object, PauliTerm):
-        assert isinstance(pauli_object.coefficient, Complex)
+        if not isinstance(pauli_object.coefficient, Complex):
+            raise ValueError("PauliTerm coefficient must be fixed and complex.")
```

Comment:
Yes, the coefficient can be a parameter. This module is a bit confusing because even when the coefficient is complex, it often asserts that the imaginary part is zero so that it can be used as a real number. I'm in two minds of how to communicate that error to the user, which is why the messages are a bit inconsistent.